docs/tutorials/getting_started/bokeh_integration.rst

```diff
@@ -0,0 +1,339 @@
+**************************
+Bokeh Integration Concepts
+**************************
+
+**Last Updated:** September 2019
+
+This tutorial introduces ``Bokeh Server`` integration concepts for Tethys developers. Two ``bokeh`` handlers will be created to demonstrate how to link Bokeh plots or widgets to Python functions in the brackground using both a plain Bokeh approach as well as a ``Param`` approach. The topics covered include:
+
+* Bokeh Server
+* Handler functions using Bokeh Widgets
+* Handler functions using Param and Panel
+
+Create a new Tethys app named bokeh_tutorial.
+
+::
+
+    $ tethys scaffold bokeh_tutorial
+
+1. Bokeh Server
+===============
+
+``Bokeh`` is an interactive visualization library for Python. ``Bokeh Server`` is a component of the ``Bokeh`` architecture. It provides a way to sync model objects from Python to the client. This is done by levering the ``Websocket`` protocol. With the addition of ``Django Channels`` to Tethys, this ability to sync python objects and frontend plots has also been integrated without the need of other components such as a ``Tornado`` server (see `Tethys Bokeh Integration documentation <../../tethys_sdk/url_maps.html#bokeh-integration>`_). This integration facilitates the linking of objects and ``Bokeh`` widgets as well as the creation of the necessary ``websocket`` and ``http`` ``consumers``.
+
+The logic for creating a Bokeh widget along with other related functionality is provided in a ``handler function``. This handler will be associated to a specific ``controller function`` where the resulting Bokeh widget will be displayed in a later step.
+
+2. Handler Functions Using Bokeh Widgets
+========================================
+
+Let's use Bokeh's sea temperature sample data to create a time series plot and link it to a slider that will provide the value to perform a rolling-window analysis on the time series. This a common example use in Bokeh's main documentation.
+
+a. Create a ``handler function`` by adding the following imports and logic to ``controller.py``.
```

Comment:
Add the step to scaffold a new app. See the GeoServer tutorial. This can be simple like in the GeoServer tutorial: http://docs.tethysplatform.org/en/stable/tutorials/geoserver/start_and_register.html#scaffold-new-app